jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketConstants.java

```diff
@@ -60,15 +62,24 @@
 
     /* Annotations */
     public static final String ON_OPEN = "OnOpen";
+    public static final String ON_CLOSE = "OnClose";
+
     public static final String IS_ANNOTATION = "isAnnotation";
 
     /* Types */
-    // For OnOpen annotation
     public static final String PATH_PARAM_ANNOTATION = "PathParam";
+
+    // For OnOpen annotation    
+    public static final Set<String> ON_OPEN_PARAM_OPT_TYPES= new HashSet<>(Arrays.asList("jakarta.websocket.EndpointConfig", "jakarta.websocket.Session"));
+    public static final Set<String> RAW_ON_OPEN_PARAM_OPT_TYPES= new HashSet<>(Arrays.asList("EndpointConfig", "Session"));
+
+    public static final Set<String> ON_CLOSE_PARAM_OPT_TYPES = new HashSet<>(Arrays.asList("jakarta.websocket.CloseReason", "jakarta.websocket.Session"));
+    public static final Set<String> RAW_ON_CLOSE_PARAM_OPT_TYPES = new HashSet<>(Arrays.asList("CloseReason", "Session"));
     
-    public final static Set<String> ON_OPEN_PARAM_OPT_TYPES= new HashSet<>(Arrays.asList("jakarta.websocket.EndpointConfig", "jakarta.websocket.Session"));
-    public final static Set<String> RAW_ON_OPEN_PARAM_OPT_TYPES= new HashSet<>(Arrays.asList("EndpointConfig", "Session"));
-    
-    public final static Set<String> RAW_WRAPPER_OBJS = new HashSet<>(Arrays.asList("String", "Boolean", "Integer", "Long", "Double", "Float"));
+    public static final Set<String> RAW_WRAPPER_OBJS = new HashSet<>(Arrays.asList("String", "Boolean", "Integer", "Long", "Double", "Float"));
     public static final Set<String> WRAPPER_OBJS = RAW_WRAPPER_OBJS.stream().map(raw -> "java.lang.".concat(raw)).collect(Collectors.toSet());
+
+    // Messages
+    public static final String INITIAL_DIAG_MSG = "Invalid parameter type. Parameter must be of type: \n- "; 
```

Comment:
When I see this message I'm not 100% sure why. Suggestion: "Invalid parameter type. When using @OnOpen parameter must be..." or @OnClose as appropriate. Consider using String.format() so the message can be "words %s more words %s" . You can see the message as a whole and just plug in the missing words.

Reply:
Yeah, makes sense, I just fixed that and pushed a new commit. Thanks!